### PR TITLE
Expanded weak passwords report

### PR DESCRIPTION
**Pull request by syntax-error752**
Needed this to be able to compare related passwords, as well as to help strengthen passwords that are only rated "good".
Added the show all checkbox as well as the password entropy calculation.

### CODE REVIEW

**Comment by syntax-error752** [comment]
Further changes needed before opening a PR.

File: src/app/tools/weak-passwords-report.component.ts [hunk 1]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
...
+                }
+                if (c.login.password.match(/[+,\/]/g) != null) {
+                    charSetSize += 3;
+                }
+                if (c.login.password.match(/[!"#$%&'()*-.:;<=>?@[\]^_\\`{|}~]/g) != null) {
+                    charSetSize += 29;
+                }
+                const passwordEntropy = Math.round(Math.log2(Math.pow(charSetSize, c.login.password.length)));
+
+                this.passwordStrengthMap.set(c.id, this.scoreKey(score, passwordEntropy));
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:89]: Pass the entropy variable just to include it in the array.

File: src/app/tools/weak-passwords-report.component.ts [hunk 2]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
                 this.passwordStrengthCache.set(cacheKey, result.score);
             }
             const score = this.passwordStrengthCache.get(cacheKey);
-            if (score != null && score <= 2) {
-                this.passwordStrengthMap.set(c.id, this.scoreKey(score));
+            if (score != null && (score <= 2 || this.showAllCheckbox)) {
+
+                let charSetSize = 0;
+                if (c.login.password.match(/[abcdefghijklmnopqrstuvwxyz]/g) != null) {
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:72]: If the password contains any lowercase letter, add 26, as there are 26 possible lowercase letters.

File: src/app/tools/weak-passwords-report.component.ts [hunk 3]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
...
+                    charSetSize += 26;
+                }
+                if (c.login.password.match(/[ABCDEFGHIJKLMNOPQRSTUVWXYZ]/g) != null) {
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:75]: If the password contains any uppercase letter, add 26, as there are 26 possible uppercase letters.

File: src/app/tools/weak-passwords-report.component.ts [hunk 4]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
...
+                    charSetSize += 26;
+                }
+                if (c.login.password.match(/\d/g) != null) {
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:78]: If the password contains any digit, add 10, as there are 10 possible digits.

File: src/app/tools/weak-passwords-report.component.ts [hunk 5]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
...
+                    charSetSize += 10;
+                }
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:80]: If the password contains I common symbol.

File: src/app/tools/weak-passwords-report.component.ts [hunk 6]
```diff
@@ -63,9 +66,31 @@ export class WeakPasswordsReportComponent extends CipherReportComponent implemen
...
+                if (c.login.password.match(/[+,\/]/g) != null) {
+                    charSetSize += 3;
+                }
+                if (c.login.password.match(/[!"#$%&'()*-.:;<=>?@[\]^_\\`{|}~]/g) != null) {
```
Comment (syntax-error752) [inline src/app/tools/weak-passwords-report.component.ts:84]: If the password contains any other symbol.

**Comment by luximus-hunter** [comment]
any progress?

**Comment by syntax-error752** [comment]
This is pretty old now, and to be honest, there's got to be a better way of running the calculations.  I'll close this for now.

**Pull request closed without merging**